server/service/integration_core_test.go

```diff
@@ -3692,6 +3692,137 @@ func (s *integrationTestSuite) TestLabels() {
 	}
 }
 
+// Sanity test to make sure fleet/labels/<all>/hosts and fleet/hosts return the same thing.
+func (s *integrationTestSuite) TestListHostsByLabel() {
```

Comment:
Thank you for adding this test! We've been hit by updating `/hosts` and missing `labels/{:id}/hosts` many times.